docs/haskell-use-cases.rst

```diff
@@ -143,6 +143,234 @@ This works for any ``haskell_binary`` or ``haskell_library`` target.
 Modules of all libraries will be loaded in interpreted mode and can be
 reloaded using the ``:r`` GHCi command when source files change.
 
+Configuring IDE integration with ghcide
+---------------------------------------
+
+rules_haskell has preliminary support for IDE integration using `ghcide`_. The
+ghcide project provides IDE features for Haskell projects through the Language
+Server Protocol. To set this up you can define a `haskell_repl`_ target that
+will collect the required compiler flags for your Haskell targets and pass them
+to `hie-bios`_ which will then forward them to ghcide.
+
+Let's set this up for the following example project::
+
+  haskell_toolchain_library(
+      name = "base",
+  )
+
+  haskell_library(
+      name = "library-a",
+      srcs = ["Lib/A.hs"],
+      deps = [":base"],
+  )
+
+  haskell_library(
+      name = "library-b",
+      srcs = ["Lib/B.hs"],
+      deps = [":base"],
+  )
+
+  haskell_binary(
+      name = "binary",
+      srcs = ["Main.hs"],
+      deps = [
+          ":base",
+          ":library-a",
+          ":library-b",
+      ],
+  )
+
+We want to configure ghcide to provide IDE integration for all these three
+targets. Start by defining a ``haskell_repl`` target as follows::
+
+  haskell_repl(
+    name = "hie-bios",
+    collect_data = False,
+    deps = [
+      ":binary",
+      # ":library-a",
+      # ":library-b",
+    ],
+  )
+
+Note, that ``library-a`` and ``library-b`` do not have to be listed explicitly.
+By default haskell_repl will include all transitive dependencies that are not
+external dependencies. Refer to the API documentation of `haskell_repl`_ for
+details.
+
+We also disable building runtime dependencies using ``collect_data = False`` as
+they are not required for an IDE session.
+
+You can test if this provides the expected compiler flags by running the
+following Bazel command and taking a look at the generated file::
+
+  bazel build //:hie-bios --output_groups=hie_bios
+
+Next, we need to hook this up to `hie-bios`_ using the `bios cradle`_. To that
+end, define a small shell script named ``.hie-bios`` that looks as follows::
+
+  #!/usr/bin/env bash
```

Comment:
This is out-of-sync with the .hie-bios script in this repo.

I guess it's on purpose (to show a simple-but-working script in the docs), but maybe we could export the full-featured script as a bazel target and let the client-side .hie-bios script just be something like

```sh
#!/bin/bash

bazel run @rules_haskell//:hie-bios //:hie-bios
```

Reply:
Indeed that was intentional to keep the documentation simple.

 > but maybe we could export the full-featured script as a bazel target

That's a very nice idea, thanks!